sensor_control/vision/symbols/src/ShoreDebug.cpp

```diff
@@ -0,0 +1,162 @@
+    if (hsv_windows)
+    {
+      Mat hsv_frame,redFrame,blueFrame,greenFrame,rtemp, rtemp2;
+      cvtColor(image,hsv_frame,CV_BGR2HSV);
+      inRange(hsv_frame,red.low,red.high,rtemp);
+      inRange(hsv_frame,red2.low,red2.high,rtemp2);
+      inRange(hsv_frame,blue.low,blue.high,blueFrame);
+      inRange(hsv_frame,green.low,green.high,greenFrame);
+      redFrame = rtemp | rtemp2;
+      imshow("blue",blueFrame);
+      imshow("green",greenFrame);
+      imshow("red",redFrame);
+    }
+    for (navigator_msgs::DockShape symbol : shapes.list) {
+        drawShape(image,symbol);
+        //poseEstimator.process(image, symbol);
+    }
+    imshow("Result",image);
+  }
+  void foundShapesCallback(const navigator_msgs::DockShapes &ds)
+  {
+    shapes = ds;
+  }
+  void initParams()
+  {
+    //Set HSV values
+    nh.getParam("hsv/red1/low/H",red.low[0]);
+    nh.getParam("hsv/red1/low/S",red.low[1]);
+    nh.getParam("hsv/red1/low/V",red.low[2]);
+    nh.getParam("hsv/red1/high/H",red.high[0]);
+    nh.getParam("hsv/red1/high/S",red.high[1]);
+    nh.getParam("hsv/red1/high/V",red.high[2]);
+
+    nh.getParam("hsv/red2/low/H",red2.low[0]);
+    nh.getParam("hsv/red2/low/S",red2.low[1]);
+    nh.getParam("hsv/red2/low/V",red2.low[2]);
+    nh.getParam("hsv/red2/high/H",red2.high[0]);
+    nh.getParam("hsv/red2/high/S",red2.high[1]);
+    nh.getParam("hsv/red2/high/V",red2.high[2]);
+
+    nh.getParam("hsv/blue/low/H",blue.low[0]);
+    std::cout << "hsv/blue/low/H " << blue.low[0] << std::endl;
+    nh.getParam("hsv/blue/low/S",blue.low[1]);
+    nh.getParam("hsv/blue/low/V",blue.low[2]);
+    nh.getParam("hsv/blue/high/H",blue.high[0]);
+    nh.getParam("hsv/blue/high/S",blue.high[1]);
+    nh.getParam("hsv/blue/high/V",blue.high[2]);
+
+    nh.getParam("hsv/green/low/H",green.low[0]);
+    nh.getParam("hsv/green/low/S",green.low[1]);
+    nh.getParam("hsv/green/low/V",green.low[2]);
+    nh.getParam("hsv/green/high/H",green.high[0]);
+    nh.getParam("hsv/green/high/S",green.high[1]);
+    nh.getParam("hsv/green/high/V",green.high[2]);
+    nh.getParam("hsv_windows",hsv_windows);
+    nh.param<std::string>("symbol_camera", camera_topic, "/right_camera/image_color");
+
+    std::cout << "Using Camera: " << camera_topic << std::endl;
+    if (hsv_windows) std::cout << "Displaying HSV Threshold Windows" << std::endl;
+  }
+ public:
+  ShoreDebug() : 
+    nh("dock_shape_shore_debug"),
+    it_(nh)
+  {
+    red = ColorThresh{Scalar(0,10, 100), Scalar(30, 255, 255)};
+    red2 = ColorThresh{Scalar(155, 10, 100), Scalar(180, 255, 255)};
+    blue = ColorThresh{Scalar(90,100,100),Scalar(150,255,255)};
+    green = ColorThresh{Scalar(30, 85, 25),Scalar(85,255,255)};
+    initParams();
+    image_sub_ = it_.subscribe(camera_topic, 1, &ShoreDebug::newFrameCallback, this);
+    foundShapesSubscriber = nh.subscribe("/dock_shapes/found_shapes", 1000, &ShoreDebug::foundShapesCallback, this);
+    namedWindow("Result",CV_WINDOW_AUTOSIZE);
```

Comment:
What we did on the sub is have the perception nodes publish debug images as they were working. So instead of having a specific debug class, the main node would be publishing out debug images and information so that you could view it in rqt